presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -83,12 +86,13 @@ public boolean isPartial()
         return hashSymbol;
     }
 
+    @JsonProperty("distinctSymbols")
     public List<Symbol> getDistinctSymbols()
     {
         if (hashSymbol.isPresent()) {
-            return ImmutableList.copyOf(Iterables.filter(getOutputSymbols(), not(hashSymbol.get()::equals)));
+            return ImmutableList.copyOf(Iterables.filter(distinctSymbols, not(hashSymbol.get()::equals)));
         }
-        return getOutputSymbols();
+        return distinctSymbols;
```

Comment:
just return distinctSymbols, the input should not contain the hashSymbol. If that's a concern, than add a checkArg for that in the constructor.